src/PerfView/MainWindow.xaml.cs

```diff
@@ -1678,14 +1678,34 @@ private void DoRename(object sender, ExecutedRoutedEventArgs e)
             var selectedFile = TreeView.SelectedItem as PerfViewFile;
             if (selectedFile == null)
                 throw new ApplicationException("No file selected.");
+            string selectedFilePath = selectedFile.FilePath;
 
             var targetPath = GetDataFileName("Rename File", false, "", null);
             if (targetPath == null)
             {
                 StatusBar.Log("Operation Canceled");
                 return;
             }
-            FileUtilities.ForceMove(selectedFile.FilePath, targetPath);
+
+            // Add a ETL suffix if the source has one.  
+            bool selectedFileIsEtl = selectedFilePath.EndsWith(".etl", StringComparison.OrdinalIgnoreCase);
+            if (selectedFileIsEtl && !Path.HasExtension(targetPath))
+                targetPath = Path.ChangeExtension(targetPath, ".etl");
+
+            // Do the move.  
+            FileUtilities.ForceMove(selectedFilePath, targetPath);
+
+            // rename all the other variations of the unmerged file
+            if (selectedFileIsEtl && targetPath.EndsWith(".etl", StringComparison.OrdinalIgnoreCase))
+            {
+                string selectedFileBase = selectedFilePath.Substring(0, selectedFilePath.Length - 4);
+                foreach (string relatedFile in System.IO.Directory.GetFiles(Path.GetDirectoryName(selectedFilePath), Path.GetFileName(selectedFileBase) + ".*"))
+                {
+                    Match m = Regex.Match(relatedFile, @"\.((clr.*)|(user.*)|(kernel.*)\.etl)");
```

Comment:
:bulb: Would be good to use `$` at the end of this expression to force it to be at the end.